src/hooks/VideoHook.js

```diff
@@ -0,0 +1,76 @@
+import {
+    useState, 
+    useEffect,
+    useReducer,
+}  from 'react';
+import axios from 'axios';
+import AppConstants from '../AppConstants';
+
+const stateEventReducer = (state, event) => {
+  switch (event.type) {
+    case 'FETCH_INIT':
+      return { 
+        ...state, 
+        isLoaded: false, 
+        isError: false 
+      };
+    case 'FETCH_SUCCESS':
+      return {
+        ...state,
+        isLoaded: true,
+        isError: false,
+        data: {
+          loadedVideos: event.payload.songs,
+        }
+      };
+    case 'FETCH_FAILURE':
+      return {
+        ...state,
+        isLoaded: false,
+        isError: true,
+      };
+    default:
+      throw new Error();
```

Comment:
Can you add a custom error message? e.g. "Unrecognized event type ${event.type}" ?